doc-surrealdb_versioned_docs/version-latest/surrealql/statements/define/access/jwt.mdx

```diff
@@ -236,3 +238,42 @@ The database token payload should at least include the following claims when use
   "db": "app_vitalsense"
 }
 ```
+
+## With `AUTHENTICATE` clause 
+
+The `AUTHENTICATE` clause allows you to define a custom expression that will be executed when the token is verified. This expression will be executed in the context of the token, allowing you to perform additional checks on the token claims before the token is accepted. If the expression returns `false`, the token will be rejected.
+
+#### Example: Using an HMAC algorithm with an `AUTHENTICATE` clause.
+
+This example defines an access policy for a user on a database using JSON Web Tokens (JWT) with the HS512 hashing algorithm. The secret key for this algorithm is provided explicitly. The `AUTHENTICATE` block outlines the authentication logic: it checks if the expiration time (`exp`) of the token is later than the current time ([`time::now()`](/docs/surrealdb/surrealql/functions/database/time#timenow)). If the token is still valid (i.e., its expiration time is in the future), the authentication process returns `true`, indicating successful authentication. If the token has expired, the process returns `false`, indicating authentication failure.
+
+```surql
+DEFINE ACCESS user ON DATABASE TYPE JWT ALGORITHM HS512 KEY "sNSYneezcr8kqphfOC6NwwraUHJCVAt0XjsRSNmssBaBRh3WyMa9TRfq8ST7fsU2H2kGiOpU4GbAF1bCiXmM1b3JGgleBzz7rsrz6VvYEM4q3CLkcO8" 
+AUTHENTICATE {
+  IF $token.exp > time::now() RETURN true;
+  RETURN false;
+};
+```
+
+#### Example: Auditing and revoking tokens
```

Comment:
I am not sure that this example should be here either. The reason is that when using third party tokens, the assumption is that the dedicated authentication service offers short-lived tokens with refresh tokens, which is superior to revocation from the security perspective. The idea for this example in the `RECORD` access type is to use it for tokens issued by SurrealDB, which does not support refresh tokens.